src/vm/callingconvention.h

```diff
@@ -1415,8 +1415,11 @@ void ArgIteratorTemplate<ARGITERATOR_BASE>::ComputeReturnFlags()
             }
 #endif
 
+#ifndef UNIX_X86_ABI
             if  (size <= ENREGISTERED_RETURNTYPE_INTEGER_MAXSIZE)
```

Comment:
If the comment is right, you should be able to tweak it here:

https://github.com/dotnet/coreclr/blob/e5994fa5507a5f08058193ff26dc3698cd2e6444/src/vm/dllimport.cpp#L3687